### PR TITLE
<TagList/> - toggle more / less tags

### PR DESCRIPTION
**Pull request by gendeld**
### 🔦 Summary

...

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://www.wix.com/pages/wix-style-react.pr_5768/

File: src/TagList/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import * as React from 'react';
 import { TagProps } from '../Tag';
 import { ButtonProps } from '../Button';
+import { TooltipProps } from '../Tooltip';
```
Comment (EgoziE) [inline src/TagList/index.d.ts:4]: change to TooltipCommonProps, you can look at CheckToggle for reference

File: src/TagList/TagList.js [hunk 2]
```diff
@@ -13,13 +15,58 @@ const tagToActionButtonSize = {
...
+      isExpanded: props.initiallyExpanded,
+    };
+  }
+
+  toggleExpanded = () =>
+    this.setState(prevState => ({ isExpanded: !prevState.isExpanded }));
+
+  renderToggleMoreButton = () => {
+    const { toggleMoreButton, tags, maxVisibleTags } = this.props;
+    const { isExpanded } = this.state;
```
Comment (EgoziE) [inline src/TagList/TagList.js:31]: Private functions should have an underscore prefix

File: src/TagList/test/TagList.spec.js [hunk 3]
```diff
@@ -24,6 +24,91 @@ describe(TagList.displayName, () => {
...
+          label: isExpanded ? 'Show Less' : `+${amountOfHiddenTags} More`,
+          tooltipProps: !isExpanded
+            ? {
+                content: 'Show More',
+              }
+            : undefined,
+        })}
+      />,
+    );
+    await driver.clickToggleMoreButton();
```
Comment (EgoziE) [inline src/TagList/test/TagList.spec.js:108]: When it comes to testing the result of an action (like a click or hover) I like to test both the before and after states.

```suggestion
    expect(await driver.toggleMoreButtonLabel()).toEqual('+1 More');
    await driver.clickToggleMoreButton();
```

File: src/TagList/index.d.ts (hunk 1 above)
Comment (gendeld) [reply src/TagList/index.d.ts:4]: That's not a type, I think TooltipCommonProps is for PropTypes (?)

**Comment by gendeld** [comment]
Separate tooltipContent from tooltip and use TooltipCommonProps

**Pull request merged**